### DISCUSSION TITLE
Help Needed: Missing Specific Wavelet Pairs in Second Order Scattering Transform Output

### DISCUSSION THREAD

**Discussion by SookyungChang** [kymatio/kymatio#1055] [Q&A] [answered]
```
J = 8  
Q = 1 
T = 0 

scattering = Scattering1D(J, input_data.shape[-1], Q, 2**T)

result = scattering(input_data)
result = result[:,1:,:] # remove the low-pass computation

print(f"The number of the first-order scattering moments: {J+1}") # 9
print(f"The number of the second-order scattering moments: {int((J+1)*J/2)}") # 36
print(f"The number of all scattering moments: {input_data.shape[1]}") # 42

```
As you can see, although the number of the first and second-order coefficients should be 9+36=45, the result is 42. By separating the first and second results, you can check there are three missing pairs, (0,1), (0,2), and (1,2) in the second-order coefficients.
(zeroth wavelet=0, first wavelet=1 and second wavelet=2)
Is there any way to also compute the three pairs?

**Comment by lostanlen** [comment, marked as answer]
Hello. First of all, you should pass `T=0`, not `2**T`.

```
>>> sc = Scattering1D(J=8, N=2**12, Q=1, T=0, out_type='dict')
```

Here's what i get:
```
>>> sc.output_size(detail=True)
(1, 9, 33)
```

```
>>> sc.meta()['n']
array([[nan, nan],
       [ 0., nan],
       [ 1., nan],
       [ 2., nan],
       [ 3., nan],
       [ 4., nan],
       [ 5., nan],
       [ 6., nan],
       [ 7., nan],
       [ 8., nan],
       [ 0.,  3.],
       [ 0.,  4.],
       [ 0.,  5.],
       [ 0.,  6.],
       [ 0.,  7.],
       [ 0.,  8.],
       [ 1.,  3.],
       [ 1.,  4.],
       [ 1.,  5.],
       [ 1.,  6.],
       [ 1.,  7.],
       [ 1.,  8.],
       [ 2.,  3.],
       [ 2.,  4.],
       [ 2.,  5.],
       [ 2.,  6.],
       [ 2.,  7.],
       [ 2.,  8.],
       [ 3.,  4.],
       [ 3.,  5.],
       [ 3.,  6.],
       [ 3.,  7.],
       [ 3.,  8.],
       [ 4.,  5.],
       [ 4.,  6.],
       [ 4.,  7.],
       [ 4.,  8.],
       [ 5.,  6.],
       [ 5.,  7.],
       [ 5.,  8.],
       [ 6.,  7.],
       [ 6.,  8.],
       [ 7.,  8.]])
```

Indeed paths `[0, 1]`, `[0, 2]`, and `[1, 2]`, are not computed.

This is because the first three wavelets at the second order are at the maximal resolution `j=0`:

```
>>> [psi['j'] for psi in sc.psi2_f]
[0, 0, 0, 1, 2, 3, 4, 5, 6]
```

and we restrict the scattering transform to `j2 > j1`
https://github.com/kymatio/kymatio/blob/main/kymatio/scattering1d/core/scattering1d.py#L87

If you need those paths, you can modify the `j` field in `sc.psi1_f` and `sc.psi2_f` after construction. I've tried, it works.

**Reply by SookyungChang** [reply]
Thanks a lot for answering my question.
Could you tell me why [0, 0, 0, 1, 2, 3, 4, 5, 6] was used, not [0, 1, 2, 3, 4, 5, 6, 7, 8]?
I mean, I was kind of expecting that if I say Q=1, the 9 wavelet filters for the first-order coefficients are also going to be used for the second-order coefficients.
I want to keep satisfying the j2 > j1 condition but hope to use all the generated 9 filters to obtain the second-order coefficients.

**Reply by lostanlen** [reply]
> Could you tell me why [0, 0, 0, 1, 2, 3, 4, 5, 6] was used, not [0, 1, 2, 3, 4, 5, 6, 7, 8]?

This is to avoid aliasing.

> I want to keep satisfying the j2 > j1 condition but hope to use all the generated 9 filters to obtain the second-order coefficients.

This is in `get_max_dyadic_subsampling`. 
https://github.com/kymatio/kymatio/blob/main/kymatio/scattering1d/filter_bank.py#L199
So far we haven't found a way to expose this to the user without compromising ease of use nor speed.
